src/main/java/cz/it4i/parallel/SimpleOstravaParadigm.java

```diff
@@ -69,8 +75,21 @@ public P_ParallelTask() {
 		}
 
 		@Override
-		public <T> T getRemoteModule(Class<T> type) {
-			return (T) Mockito.mock(type, new P_InvocationHandler<T>(type));
+		public <T extends Command> T getRemoteCommand(Class<T> type) {
+
+			T mockedCommand = mock(type, CALLS_REAL_METHODS);
+
+			doAnswer(new Answer<T>() {
```

Comment:
Answer should have type Void instead of T - it is return type for specific code. 
Error was introduced in previous version with InvocationHandler.

Reply:
Removed within your enhancement.